### PR TITLE
feat(core): Made SimpleChange fields readonly.

### PR DESCRIPTION
**Pull request by trshafer**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
> not really possible to add tests for readonly. They will create compile errors.
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[?] Feature
[ ] Code style update (formatting, local variables)
[?] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
Current behavior is ability to overwrite the simple changes values.

**What is the new behavior?**
New behavior is to mark simple change values as readonly.

**Does this PR introduce a breaking change?** (check one with "x")
```
[?] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
This is a low profile API.

**Other information**:

### CODE REVIEW

**Comment by trshafer** [comment]
Hey Angular team, any update on this?

**Comment by splincode** [comment]
@trshafer What is the status of the task?

**Comment by trshafer** [comment]
@splincode I haven't done anything with it as it's waiting for Angular support.

**Review by mhevery — APPROVED** [review]

**Comment by splincode** [comment]
@trshafer please, you need fix conflicts

**Comment by mhevery** [comment]
http://test/OCL:242758278:BASE:242758538:1554850975318:c0d3fe6d

**Comment by mhevery** [comment]
http://test/OCL:260226215:BASE:260226249:1564180391709:43c8c4e

File: packages/elements/src/component-factory-strategy.ts [hunk 1]
```diff
@@ -231,7 +231,7 @@ export class ComponentNgElementStrategy implements NgElementStrategy {
     // previousValue and isFirstChange.
     const pendingChange = this.inputChanges[property];
     if (pendingChange) {
-      pendingChange.currentValue = currentValue;
+      (pendingChange as{currentValue: any}).currentValue = currentValue;
```
Comment (andrewseguin) [inline packages/elements/src/component-factory-strategy.ts:234]: Seems like it shouldn't be marked as `readonly` if this is a valid use case. Changing this to `any` doesn't seem right

Comment (mhevery) [reply packages/elements/src/component-factory-strategy.ts]: It is `readonly` for public API, but internally we have to set this object up.

**Review by mhevery — APPROVED** [review]
reviewed-for: global-approvers

**Comment by AndrewKushnir** [comment]
[Global Presubmit](https://test.corp.google.com/OCL:343435129:BASE:343435239:1605853538869:616640d6).

**Comment by mhevery** [comment]
This is quite breaking in googl3 as people do write to this in tests. Not sure this is worth the migration trouble.

**Comment by mhevery** [comment]
This is a good PR with good intention, but trying to land this at Google shows that a lot of teams write to these properties in their tests. The practice is sufficiently wide spread that we don't think the effort justifies the benefits and as such we will be closing this PR.

**Pull request closed without merging**